[lightgbm] Add Threadless Feature

dsmith111 opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Adding the threadless option/feature listed here:
https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html#build-threadless-version-not-recommended

To increase compatibility with projects which do not use OpenMP.

dsmith111 committed:
Add threadless, disable openmp option

dsmith111 committed:
Update lightgbm baseline

dsmith111 committed:
Update lightgbm baseline

dsmith111 committed:
Merge branch 'smithdavi/lightgbm-feature-threadless' of https://github.com/dsmith111/vcpkg into smithdavi/lightgbm-feature-threadless

dsmith111 committed:
Remove comment

dsmith111 committed:
update baseline

jimwang118 commented:
The feature test passed with following triplets:
```
x64-windows
x64-windows-static
```

dg0yt commented:
Do we really want this to be implemented with a unique feature "threadless"?
The general pattern in other ports is to have a feature `openmp`.
For this package, it is even recommended to use OpenMP, so `openmp` should be a default feature if adopted.

dsmith111 committed:
Change naming from threadless to openmp;

dsmith111 committed:
hash update

dsmith111 commented:
> Do we really want this to be implemented with a unique feature "threadless"? The general pattern in other ports is to have a feature `openmp`. For this package, it is even recommended to use OpenMP, so `openmp` should be a default feature if adopted.

Two asks here.

> Do we really want this to be implemented with a unique feature "threadless"? The general pattern in other ports is to have a feature openmp.
The package referred to it as threadless is all. Updated the naming to "openmp"

> For this package, it is even recommended to use OpenMP, so `openmp` should be a default feature if adopted.
Yes, it is recommended and the original threadless feature was _not_ a default feature. So using openmp was the default behavior.

Updated the naming from threadless to openmp; flipped the default behavior to be on.

dsmith111 committed:
Update option setting.

dsmith111 committed:
update hash
